src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int MAXIMUM_CAPACITY = Integer.MAX_VALUE / 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int VALUE_FOR_INCREASE = 2;
+    private Node<K, V>[] entryTable;
+    private int size;
+    private int threshold;
+    private final float loadFactor;
+
+    public MyHashMap() {
+        this(DEFAULT_INITIAL_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    private MyHashMap(int initialCapacity, float loadFactor) {
+        if (initialCapacity < 0) {
+            throw new IllegalArgumentException("Wrong initial capacity: " + initialCapacity);
+        }
+
+        if (initialCapacity > MAXIMUM_CAPACITY) {
+            initialCapacity = MAXIMUM_CAPACITY;
+        }
+
+        if (loadFactor <= 0 || Float.isNaN(loadFactor)) {
+            throw new IllegalArgumentException("Wrong load factor: " + loadFactor);
+        }
+
+        int capacity = 1;
+        while (capacity < initialCapacity) {
+            capacity++;
+        }
+
+        entryTable = new Node[capacity];
+        this.loadFactor = loadFactor;
+        threshold = (int) (capacity * loadFactor);
+    }
```

Comment:
I would not overcomplicate it at this point
let's have only constructor with no parameters?

this constructor is private as well - we can make a single simple default constructor